eclipse-theia/theia
# replace json theia extension with vscode one

#### What it does


Fixes: #7981 
Fixes: #3983 

- [x] fix 7981, fix 3983: replace json theia extension with vscode one
- [x] document breaking changes
- [x] deprecate contributing language support from Theia extension

Please see [CHANGELOG](https://github.com/eclipse-theia/theia/blob/ak/json_vscode/CHANGELOG.md#1_4_0_replace_json) for breaking changes.

#### How to test


- verify that smartness available in settings (user, workspace), debug and task files as before
- verify that user data as settings and keymaps are preserved
- verify that jsonValidation contribution point from VS Code extension is respected
- verify `How to test` from https://github.com/eclipse-theia/theia/pull/7225

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
